### ISSUE TITLE
Possible SCII_SPOILED_CHILD_INTERFACE_IMPLEMENTOR false positive

### ISSUE THREAD

**Issue by efenderbosch** [mebigfatguy/fb-contrib#173]
We are developing using the Play Framework. Play uses Guice for DI. In order to inject Akka Actors, Play has a Java 8 interface with default methods, ``AkkaGuiceSupport``.

It is supposed to be used like this:

```java
import com.google.inject.AbstractModule;
import play.libs.akka.AkkaGuiceSupport;
public class MyModule extends AbstractModule implements AkkaGuiceSupport {
    protected void configure() {
        // bindActor is a default method in AkkaGuiceSupport
        bindActor(MyActor.class, "myActor");
    }
}
```
  
During compilation, we get a ``SCII_SPOILED_CHILD_INTERFACE_IMPLEMENTOR`` error from fb-contrib.

Reading the description, it doesn't sound like this is an intended trigger. Is this because ``AkkaGuiceSupport`` is a Java 8 Interface with only default methods?

For now we are simply adding ``@SuppressFBWarnings("SCII_SPOILED_CHILD_INTERFACE_IMPLEMENTOR")`` to our Guice modules that require Akka Actor injection. Is this a false positive, or should we simply continue to use the annotation to ignore this error?

Source code for Guice's Abstract Module:
https://github.com/google/guice/blob/4.0/core/src/com/google/inject/AbstractModule.java

Source code for Play's AkkaGuiceSupport:
https://github.com/playframework/playframework/blob/2.5.x/framework/src/play/src/main/java/play/libs/akka/AkkaGuiceSupport.java

**Comment by mebigfatguy** [comment]
Thanks for the report.. Given that AkkaGuiceSupport doesn't declare any interface methods that are overwritten by AbstractModule, This appears to be a FP. This is indeed likely to do with some confusion with default methods. I'll take a look.

**Comment by mebigfatguy** [comment]
thanks, was indeed default methods. fixed in head.

**Comment by efenderbosch** [comment]
Has this been released? Just encountered another instance using 6.8.4.

```java
private enum AttributeDefinitionNameComparator implements Comparator<AttributeDefinition> {
        INSTANCE;
        @Override
        public int compare(AttributeDefinition o1, AttributeDefinition o2) {
            // yes, i know this isn't null safe
            return o1.getAttributeName().compareTo(o2.getAttributeName());
        }
    }
```

**Comment by mebigfatguy** [comment]
thanks for the report. yes the above should have been released with 6.8.4

this issue is different. It's that an interface (Comparator) defines a method that's in Object (equals). 

So Object methods showing up in interfaces need to be screened out as well. will fix tonite..
